### ISSUE TITLE
Cursor problems in high contrast theme

### ISSUE THREAD

**Issue by JonathanMEdwards** [Microsoft/vscode#3605]
The line cursor becomes thinner in the high contrast theme.
A line cursor disappears when next to a bracket character, obscured by the box rendered around the bracket.
Flashing block cursor is not very readable because it doesn't have a good inverse-video algorithm, so the underlying character becomes unreadable half the time.

**Comment by bgashler1** [comment]
@JonathanMEdwards thanks for reporting this.  Can you clarify on the flashing block cursor.  What steps did you take to enter the block cursor mode?  Could show a screenshot of what you're seeing?

**Comment by JonathanMEdwards** [comment]
settings.json editor.cursorStyle: "block"

https://vimeo.com/157368124

**Comment by leonardosnt** [comment]
Me too

**Comment by leonardosnt** [comment]
@bgashler1 I'm using "editor.cursorStyle": "block", and is not working with "High contrast" theme, is just with High contrast, with others he is working fine.

Sorry for my poor english.

**Comment by bgashler1** [comment]
Thanks everyone.  I'll be sure to get a fix in for this.

**Comment by JonathanMEdwards** [comment]
How about the line cursor being thinner than in other themes?

**Comment by misolori** [comment]
Not able to repro any of these, @JonathanMEdwards can you let me know if you're still seeing these issues in the latest version? Here's what I'm seeing:

> The line cursor becomes thinner in the high contrast theme.

![image](https://user-images.githubusercontent.com/35271042/45389558-1f234e80-b5d1-11e8-9def-fdc322f521c3.png)


> A line cursor disappears when next to a bracket character, obscured by the box rendered around the bracket.

![image](https://user-images.githubusercontent.com/35271042/45389650-63165380-b5d1-11e8-8792-7d610c0eea0b.png)


> Flashing block cursor is not very readable because it doesn't have a good inverse-video algorithm, so the underlying character becomes unreadable half the time.

![gif](https://user-images.githubusercontent.com/35271042/45389766-c3a59080-b5d1-11e8-9940-6d23ebb5802c.gif)

**Comment by JonathanMEdwards** [comment]
Right, these problems seem to have gone away.

**Comment by misolori** [comment]
@JonathanMEdwards thanks for verifying!
